couchdb/src/test/java/org/hibernate/ogm/datastore/couchdb/test/crud/EntityWithByteArrayTest.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.couchdb.test.crud;
+
+import java.util.Arrays;
+import java.util.Random;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.annotations.GenericGenerator;
+import org.hibernate.ogm.utils.OgmTestCase;
+import org.hibernate.ogm.utils.TestForIssue;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertNotNull;
+
+/**
+ * @author Hardy Ferentschik
+ */
+public class EntityWithByteArrayTest extends OgmTestCase {
```

Comment:
> Yes, BuiltInTypeTest.testTypesSupport is pretty big and I think it would be nice to refactor it.
We could probably separate each assert in it's own unit test.

Separating the assertions is probably a good start. One can still use a single test entity, but at least separate the different assertions.

> That said, I think for the issue you are working at the moment it would be better to be consistent with what we have now.

I don't see how consistency is broken. IMO are there two different concerns. One is the issue I was working on. I think it deserves it's own dedicated test case referenced by `@TestForIssue`. It also should live in the CouchDB module. 

I also agree that we need a "tck" test in case we want to make sure this case works for all datastores, but most likely this should be a new issue. And even if we address it within the same issue, I'd argue that there is no harm in keeping two tests verifying the same thing. The tck and the actual module test fulfill different purposes imo. We have the same in Validator. We have the BV TCK and the actual tests in the engine as well. There is a huge overlap.